src/components/layout/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,49 @@
+"use client";
+import { useState } from "react";
+import styles from "./Sidebar.module.scss";
+import { LuChevronLeft, LuChevronRight } from "react-icons/lu";
+
+import MenuProfil from "@/components/ProfileComponents/Menu/Menu";
+
+interface SidebarProps {
+  // role: "user" | "artisan";
+  isOpen: boolean; // Paramètre pour contrôler l'état ouvert/fermé
+  toggleSidebar: () => void; // Fonction pour gérer le toggle
+  children: React.ReactNode; // Pour injecter du contenu
+  location: "right" | "left";
+}
+
+export default function Sidebar({
+  isOpen,
+  toggleSidebar,
+  children,
+  location = "left",
+}: SidebarProps) {
+  // const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+
+  // const toggleSidebar = () => {
+  //   setIsSidebarOpen(!isSidebarOpen);
+  // };
+
+  return (
+    <div
+      className={`
+        ${styles.sidebarContainer} 
+        ${isOpen ? styles.sidebarOpen : styles.sidebarClosed}
+        ${location === "right" ? styles.sidebarRight : styles.sidebarLeft}
```

Comment:
${location === "right" ? styles.sidebarRight : styles.sidebarLeft}
```suggestion
        ${location === "right" ? styles.rightSidebar : styles.leftSidebar}
```